.circleci/config.yml

```diff
@@ -536,6 +553,7 @@ commands:
       - save_test_results
       - send_coverage
       - persist_coverage
+      - notify-slack
```

Comment:
For testing purposes, this is what the Slack notifications looked like when notified separately on each job's success: https://oicr.slack.com/archives/C0151UDNR9R/p1712088975956109

Things to consider:

- If we choose to notify Slack only on failed jobs, these messages probably won't occur too frequently; nightly runs will probably succeed most of the time
- I don't think it'd be possible to get failed notifications from the build, unit tests, and integration tests all at once. If the build fails/notifies, for example, the dependent jobs (i.e. all the tests) should be skipped and can't notify. Worst case scenario is if the build passes, and all the unit/integration tests fail?
- Also just noticed that some integration tests produce multiple notifications (e.g. 3 for confidential-tool/workflow-tests, 2 for non-confidential-tests); appears to be due to the parallelism

Reply:
> * If we choose to notify Slack only on failed jobs, these messages probably won't occur too frequently; nightly runs will probably succeed most of the time

I think notifying on failed jobs is better since those are the actionable ones. That said, we can run into the problem that it the notifications break ... 

Especially with the parallel notifications, 

> some integration tests produce multiple notifications 

I think there are too many notifications.


> Worst case scenario is if the build passes, and all the unit/integration tests fail?

If I understand correctly, we'd get messages for all the failures if we were notifying on failures only. 
I think that's ok. I think numerous failure messages in a worse case scenario is better than numerous success messages (which may lead us to ignore messages)

Open to ideas